hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataInputNavigableJsonAdapter.java

```diff
@@ -131,7 +132,9 @@ public int read() throws IOException {
 
         @Override
         public int read(char[] cbuf, int off, int len) throws IOException {
-            final CharsetDecoder decoder = DECODER_THREAD_LOCAL.get();
+            final CharsetDecoder decoder = Thread.currentThread() instanceof OperationThread
+                    ? DECODER_THREAD_LOCAL.get()
+                    : Bits.UTF_8.newDecoder();
```

Comment:
Creating a new decoder on each call to `read()` (for non-operation threads) seems unnecessarily expensive, especially since there is no danger (at least I think given how/where we use `NavigableJsonAdapter`) in reusing a single instance decoder instance per `UTF8Reader`.